envoy/stream_info/stream_info.h

```diff
@@ -604,5 +604,43 @@ class StreamInfo {
   virtual absl::optional<uint32_t> attemptCount() const PURE;
 };
 
+// An enum representation of the Proxy-Status error space.
+enum ProxyStatusError {
```

Comment:
Please use `enum class` for an extra bit of type safety.

Reply:
Done.